src/cargo/ops/cargo_compile.rs

```diff
@@ -92,8 +99,10 @@ pub enum CompileFilter<'a> {
 
 pub fn compile<'a>(ws: &Workspace<'a>, options: &CompileOptions<'a>)
                    -> CargoResult<ops::Compilation<'a>> {
-    for key in try!(ws.current()).manifest().warnings().iter() {
-        try!(options.config.shell().warn(key))
+    if let Some(root_package) = ws.current_opt() {
```

Comment:
Random drive-by comment, but could this actually iterate over all members of the workspace? That's the intention here, at least